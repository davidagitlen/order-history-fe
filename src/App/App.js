import React, { Component } from 'react';
import Orders from '../Orders/Orders';
import OrderForm from '../OrderForm/OrderForm';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      orders : [],
      error : ''
    }
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/purchases')
      .then(res => res.json())
      // .then(data => console.log(data))
      .then(data => this.setState({orders : data, error : ''}))
      .catch(err => this.setState({error : err.message}))
      // .catch(err => console.log(err))
  }



  render() {
    return (
      <div className="App">
        <header>
          <h1 className='app-title'>My Order History</h1>
          <div className='purchase-form'>
            <OrderForm />
          </div>
        </header>
        <div className='purchase-container'>
          <Orders orderArray={this.state.orders} />
        </div>
      </div>
    );
  }
}

export default App;
