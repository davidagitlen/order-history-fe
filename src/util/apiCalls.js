export const getOrderHistory = () => {
  return fetch('http://localhost:3001/api/v1/purchases')
    .then(res => res.json())
}

export const postPurchase = newPurchase => {
  const options = {
    method: 'POST',
    body: JSON.stringify(newPurchase),
    headers: {
      'Content-Type': 'application/json'
    }
  }

  return fetch('http://localhost:3001/api/v1/purchases', options)
    .then(res => res.json())
}

export const deletePurchase = id => {
  return fetch(`http://localhost:3001/api/v1/purchases/${id}`, { method: 'DELETE' })
    .then(res => res.json())
}