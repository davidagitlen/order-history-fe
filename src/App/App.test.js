import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { getOrderHistory, postPurchase, deletePurchase } from '../util/apiCalls.js';

jest.mock('../util/apiCalls.js');

describe('App', () => {
  let wrapper; 

  beforeEach(() => {
    getOrderHistory.mockImplementation(() => {
      return Promise.resolve([{id: 1, image: 'broken-link.png', name: 'Blah', description: 'blah', price: 100}])
    });
    postPurchase.mockImplementation(() => {
      return Promise.resolve({id: 2, image: 'broken-link.png', name: 'Two', description: 'two', price: 200})
    });
    wrapper = shallow(<App />);
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state with data after mounting', () => {
    shallow(<App />);
    expect(getOrderHistory).toHaveBeenCalled();
  });

  it.skip('should add a purchase when addNewPurchase is called', () => {

  })
})
