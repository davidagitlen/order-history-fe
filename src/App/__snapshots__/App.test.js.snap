// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <header>
    <h1
      className="app-title"
    >
      My Order History
    </h1>
    <div
      className="purchase-form"
    >
      <OrderForm
        addNewPurchase={[Function]}
      />
    </div>
  </header>
  <div
    className="purchase-container"
  >
    <Orders
      orderArray={
        Array [
          Object {
            "description": "blah",
            "id": 1,
            "image": "broken-link.png",
            "name": "Blah",
            "price": 100,
          },
        ]
      }
      removePurchase={[Function]}
    />
  </div>
</div>
`;
